CreatubblesAPIClient/Sources/CreationUploadSession/CreationUploadSession.swift

```diff
@@ -23,6 +23,23 @@
 
 import UIKit
 
+// REFACTORING NOTES:
+// Sebastian: From my perspective refactoring effort should start with discussion on how should the upload work from user perspective: reporting progress (multi step?), reporting errors (currently just X - not useful), possibly error recovery options - we should know what is needed from the user perspective so that we can design a minimal upload API satisfying those requirements
+// 0. current setup of upload classes is constrained by the fact, that it's located in the API client - it's higher level logic and we could consider moving it
```

Comment:
I think we need to re-think how we connect to our API. I don't like the fact that we have duplicated model classes (for starters), but this is just another piece of evidence towards changing that.

FWIW In rome next week we're likely going to be building a new standalone teacher app, perhaps it'll be best to NOT use the APIClient there and see what we end up with.

Thanks for this write up @sebastianewak :)

Reply:
When starting a new app it would make sense to start fresh. Still, I imagine bits and pieces from the existing client might prove useful (not necessarily in exactly the existing form).